src/WebJobs.Script/Config/ApplicationInsightsLoggerFilterOptionsSetup.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using Microsoft.Azure.WebJobs.Logging.ApplicationInsights;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    internal class ApplicationInsightsLoggerFilterOptionsSetup : IConfigureOptions<LoggerFilterOptions>
+    {
+        private readonly IEnvironment _environment;
+
+        internal static readonly AsyncLocal<bool> FilterApplicationInsightsFromWorker = new AsyncLocal<bool>();
+
+        public ApplicationInsightsLoggerFilterOptionsSetup(IEnvironment environment)
+        {
+            _environment = environment;
+        }
+
+        public void Configure(LoggerFilterOptions options)
+        {
+            // Out-of-proc workers have the option of handling App Insights by themselves. If this is the case, they can
+            // set this AsyncLocal to true to indicate the log should be skipped.
+            if (!_environment.IsDotNetInProc())
+            {
+                options.AddFilter<ApplicationInsightsLoggerProvider>((_, _) => !FilterApplicationInsightsFromWorker.Value);
```

Comment:
Given the pains we have had with provider-specific filters, I do wonder if this will also prevent the non-provider filters from being applied?

Can we verify if when this is `True`, that other filters will still be applied?

Reply:
Yes, I can add a test for this. To be specific, it's that b/c this is specific, it gets picked and other filters are ignored so if a worker does not have a capability, suddenly something generic like this in host.json would be ignored?

```
{
  "Logging": {
    "LogLevel": {
      "Function.Function1.User": "None" 
    }
  }
}
```

Reply:
Okay -- good call here as it did completely interact with the other filters in non-intuitive ways. I've gone ahead and moved the check to an `ITelemetryProcessor` now, meaning that only App Insights will see this and all filters remain as-is.